src/engine/test/source/builder/builders/opBuilderHelperNetInfoAddress_test.cpp

```diff
@@ -0,0 +1,615 @@
+    });
+
+    auto op =
+        bld::opBuilderHelperSaveNetInfoIPv4(tuple)->getPtr<Term<EngineOp>>()->getFn();
+    result::Result<Event> result = op(event1);
+    ASSERT_TRUE(result);
+
+    t.join();
+    close(serverSocketFD);
+}
+
+TEST(opBuilderHelperNetInfoTest, Correct_execution_without_broadcast_netmask)
+{
+    auto tuple = std::make_tuple(std::string {"/field"},
+                                 std::string {"saveNetInfoIPv4"},
+                                 std::vector<std::string> {"$agent.id",
+                                                           "$event.original.ID",
+                                                           "$event.original.iface.name",
+                                                           "$event.original.iface.IPv4"});
+
+    auto event1 = std::make_shared<json::Json>(
+        R"(
+        {
+            "agent":
+            {
+                "id": "021"
+            },
+            "event":
+            {
+                "original":
+                {
+                    "ID": "123456",
+                    "iface":
+                    {
+                        "name": "iface_name",
+                        "IPv4":
+                        {
+                            "address":
+                            [
+                                "192.168.10.15",
+                                "192.168.11.16",
+                                "192.168.100.16"
+                            ]
+                        }
+                    }
+                }
+            }
+        })");
+
+    // Create the endpoint for test
+    const int serverSocketFD {testBindUnixSocket(TEST_STREAM_SOCK_PATH, SOCK_STREAM)};
+    ASSERT_GT(serverSocketFD, 0);
+
+    std::thread t([&]() {
+        const int clientRemote {testAcceptConnection(serverSocketFD)};
+        ASSERT_STREQ(
+            testRecvString(clientRemote, SOCK_STREAM).c_str(),
+            "agent 021 netaddr save 123456|iface_name|0|192.168.10.15|NULL|NULL");
+        testSendMsg(clientRemote, "ok");
+        close(clientRemote);
+
+        const int SecondclientRemote {testAcceptConnection(serverSocketFD)};
+        ASSERT_STREQ(
+            testRecvString(SecondclientRemote, SOCK_STREAM).c_str(),
+            "agent 021 netaddr save 123456|iface_name|0|192.168.11.16|NULL|NULL");
+        testSendMsg(SecondclientRemote, "ok");
+        close(SecondclientRemote);
+
+        const int ThirdclientRemote {testAcceptConnection(serverSocketFD)};
+        ASSERT_STREQ(
+            testRecvString(ThirdclientRemote, SOCK_STREAM).c_str(),
+            "agent 021 netaddr save 123456|iface_name|0|192.168.100.16|NULL|NULL");
+        testSendMsg(ThirdclientRemote, "ok");
+        close(ThirdclientRemote);
+    });
+
+    auto op =
+        bld::opBuilderHelperSaveNetInfoIPv4(tuple)->getPtr<Term<EngineOp>>()->getFn();
+    result::Result<Event> result = op(event1);
+    ASSERT_TRUE(result);
+
+    t.join();
+    close(serverSocketFD);
```

Comment:
Please, do the `join`'s before the `assert`'s, (please do it on every test).

Reply:
got it